lib/service/service.go

```diff
@@ -3863,6 +3864,13 @@ func (process *TeleportProcess) initProxyEndpoint(conn *Connector) error {
 				utils.GRPCServerStreamErrorInterceptor,
 				proxyLimiter.StreamServerInterceptor,
 			),
+			grpc.KeepaliveParams(keepalive.ServerParameters{
```

Comment:
For now it is the only service using this gRPC server, if/when a new service is added in the future we could considerably increase this or remove it.

But I think it is useful to have a connection timeout now while we know there are teleport clients in the wild which will not close their connections. Some other customer is bound to run into this issue and it's very helpful to have it mostly solved with just a proxy upgrade.

Context for this is that while we can implement a timeout in the individual RPC or with an interceptor, if the client doesn't explicitly close the `grpc.ClientConn`, it will keep the TCP connection open. This consumes a file descriptor on the server and there will be TCP keep-alive frames going back and forth indefinitely. Currently, failing clients will open 4 connections per minute and never close them.

Some alternatives would be to:
1. Implement a connection limiter at the tcp layer. The limit would have to be quite high and not configurable per RPC, and I'd be worried about breaking other services.
2. Save incoming tls.Conns somewhere and make them accessible to the join service handler so that they can be forcibly closed there when they are detected to be misbehaving. To me this seems more hacky than this `MaxConnectionIdle`.
3. Don't close the connections on the server, rely on the clients. It's hard to get customers currently facing this issue to deploy a fix to all nodes.

I've added a more descriptive comment here for now.